apps/vmq_server/test/vmq_mqtt_opts_SUITE.erl

```diff
@@ -0,0 +1,191 @@
+-module(vmq_mqtt_opts_SUITE).
+
+-include_lib("vmq_commons/include/vmq_types.hrl").
+
+-compile(export_all).
+-compile(nowarn_export_all).
+
+%% ===================================================================
+%% common_test callbacks
+%% ===================================================================
+init_per_suite(Config) ->
+  cover:start(),
+  [{ct_hooks, vmq_cth} | Config].
+
+
+end_per_suite(_Config) ->
+  _Config.
+
+init_per_testcase(_Case, Config) ->
+  vmq_test_utils:setup(),
+  vmq_server_cmd:set_config(allow_anonymous, true),
+  vmq_server_cmd:set_config(retry_interval, 10),
+  vmq_server_cmd:set_config(max_client_id_size, 1000),
+  vmq_server_cmd:listener_start(1888, [{allowed_protocol_versions, "3,4,5"}]),
+  enable_on_publish(),
+  enable_on_subscribe(),
+  Config.
+
+end_per_testcase(_, Config) ->
+  disable_on_publish(),
+  disable_on_subscribe(),
+  vmq_test_utils:teardown(),
+  Config.
+
+all() ->
+  [
+    {group, mqttv4}
+  ].
+
+groups() ->
+  V4Tests =
+    [qos1_persist_test,
+     qos1_non_persist_test,
+     qos1_retry_test,
+     qos1_non_retry_test],
+  [
+    {mqttv4, [], V4Tests}
+  ].
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%%% Actual Tests
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+qos1_persist_test(Cfg) ->
+  Connect = packet:gen_connect(vmq_cth:ustr(Cfg) ++ "qos1-persist-test", [{keepalive,60}, {clean_session, false}]),
+  Connack1 = packet:gen_connack(0),
+  Connack2 = packet:gen_connack(true, 0),
+  Disconnect = packet:gen_disconnect(),
+  Subscribe = packet:gen_subscribe(109, "qos1/persistence/test", 1, false, false),
+  Suback = packet:gen_suback(109, 1),
+  Publish = packet:gen_publish("qos1/persistence/test", 1, <<"persist-message">>, [{mid, 1}]),
+  {ok, Socket} = packet:do_client_connect(Connect, Connack1, []),
+  enable_on_publish(),
```

Comment:
This should be auth_on_publish and same for auth_on_subscribe.

Reply:
Why?